packages/react/src/__tests__/ReactElementValidator-test.js

```diff
@@ -388,36 +331,28 @@ describe('ReactElementValidator', () => {
       }
     }
 
-    ReactTestUtils.renderIntoDocument(React.createElement(Component));
-    ReactTestUtils.renderIntoDocument(
-      React.createElement(Component, {prop: 42}),
-    );
-
-    if (__DEV__) {
-      expect(console.error.calls.count()).toBe(2);
-      expect(console.error.calls.argsFor(0)[0]).toBe(
-        'Warning: Failed prop type: ' +
-          'The prop `prop` is marked as required in `Component`, but its value ' +
-          'is `undefined`.\n' +
-          '    in Component',
+    expect(() => {
+      ReactTestUtils.renderIntoDocument(React.createElement(Component));
+      ReactTestUtils.renderIntoDocument(
+        React.createElement(Component, {prop: 42}),
       );
-
-      expect(console.error.calls.argsFor(1)[0]).toBe(
-        'Warning: Failed prop type: ' +
-          'Invalid prop `prop` of type `number` supplied to ' +
-          '`Component`, expected `string`.\n' +
-          '    in Component',
+    }).toWarnDev([
+      'Warning: Failed prop type: ' +
+        'The prop `prop` is marked as required in `Component`, but its value ' +
+        'is `undefined`.\n' +
+        '    in Component',
+      'Warning: Failed prop type: ' +
+        'Invalid prop `prop` of type `number` supplied to ' +
+        '`Component`, expected `string`.\n' +
+        '    in Component',
+    ]);
+
+    // Should not error for strings
+    expect(() => {
```

Comment:
Same; I would expect any warning that isn’t inside a protected block to fail immediately.

Reply:
I think this was just me being silly.

Reply:
To be clear, I expect that with current infra it fails at the end of the test.

However, I propose that `toWarnInDev` acts as a boundary, and we change `console.error`/`console.warn` to **immediately throw** outside of that boundary. This way we get the error immediately at the right call site instead of at the end of the test.